idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3997 @@
+                sqz = 10*sqx # Heuristic
+                
+                sxflux = {}
+                for j in self.comps:
+                    #ssf(self.conc_ref[t,iz,j],sy_def*1e-4/(sR*sT))
+                    
+                    if self.xflux_x0[t,iz,j].is_reference():
+                        sxflux0 = gsf(self.xflux_x0[t,iz,j].referent, default=1e-2)
+                    else:
+                        sxflux0 = sgsf(self.xflux_x0[t,iz,j], 1e-2)
+                    cst(self.xflux_x0_eqn[t,iz,j], sxflux0)
+                    if not self.Dconc_x0[t,iz,j].is_reference():
+                        ssf(self.Dconc_x0[t,iz,j], sxflux0*sLx/sD)
+                    
+                    if self.xflux_x1[t,iz,j].is_reference():
+                        sxflux1 = gsf(self.xflux_x1[t,iz,j].referent, default=1e-2)
+                    else:
+                        sxflux1 = sgsf(self.xflux_x1[t,iz,j], 1e-2)
+                    cst(self.xflux_x1_eqn[t,iz,j],sxflux1)
+                    
+                    if not self.Dconc_x1[t,iz,j].is_reference():
+                        ssf(self.Dconc_x1[t,iz,j], sxflux1*sLx/sD)
+                    
+                    sxflux[j] = min(sxflux0, sxflux1)
+                
+                for ix in self.ixnodes:
+                    sP = sgsf(self.pressure[t,ix,iz],1e-4)
+                    sV = sR*sT/sP
+                    
+                    sDT = sgsf(self.Dtemp[t,ix,iz],sqx*sLx/sk)
+                    sH = sgsf(self.enth_mol[t,ix,iz],sH)
+                    cst(self.enth_mol_eqn[t,ix,iz],sH)
+                    
+                    if self.config.has_holdup:
+                        sU = sgsf(self.int_energy_mol[t,ix,iz],sH)
+                        cst(self.int_energy_mol_eqn[t,ix,iz],sU)
+                        
+                        s_rho_U = sgsf(self.int_energy_density[t,ix,iz],sU/sV)
+                        cst(self.int_energy_density_eqn[t,ix,iz],s_rho_U)
+                        
+                        s_rho_U_solid = sgsf(self.int_energy_density_solid[t,ix,iz],
+                                             1/(self.solid_heat_capacity.value
+                                                *self.solid_density.value*sDT))
+                        cst(self.int_energy_density_solid_eqn[t,ix,iz],s_rho_U_solid)
+                    
+                    cst(self.mole_frac_eqn[t,ix,iz],1)
+                    cst(self.energy_balance_solid_eqn[t,ix,iz],sqx*sLy*sLz)
+                    
+                    for j in self.comps:
+                        sy = sgsf(self.mole_frac_comp[t,ix,iz,j],sy_def)
+                        cst(self.conc_eqn[t,ix,iz,j],sy*sP)
+                        ssf(self.Dconc[t,ix,iz,j],sxflux[j]*sLx/sD)
+                        
+                        cst(self.material_balance_eqn[t,ix,iz,j],
+                            sxflux[j]*sLy*sLz)
+                        
+               
+@declare_process_block_class("SocTriplePhaseBoundary")
```

Comment:
With three of these classes in a single module, this is ends up being rather long. I would suggest you consider breaking this into separate modules for each class to help keep things shorter and easier to work throguh.

Reply:
@dallan-keylogic if you do this, I suggest creating a like soc_common for the common content then one module for each submodel.  Maybe that's obvious, just pointing out that it doesn't need to be much work.